lib/perl/Genome/SoftwareResult/User.t

```diff
@@ -68,6 +72,7 @@ my $new_sr = $class->with_registered_users(
             callback => \&_newly_created_callback,
             %users_hash
         );
+UR::Context->commit();
```

Comment:
In other, maybe better words...

If you have two things trying to `_get_or_create_with_lock` in `precommit` and the lock isn't removed until `commit`.  Isn't that a deadlock?  Tests didn't catch it because they're calling commit on the wrong context?